src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/NuGetProjectService.cs

```diff
@@ -72,6 +74,17 @@ NuGetInstalledPackage ToNuGetInstalledPackage(PackageReference packageReference,
                 string requestedRange =
                     packageReference.AllowedVersions.OriginalString // most packages
                     ?? packageReference.AllowedVersions.ToShortString();
+
+                if (versionRange.MinVersion == null)
+                {
+                    RestoreLogMessage nug1604Msg = RestoreLogMessage.CreateWarning(
+                       code: NuGetLogCode.NU1604,
+                       message: string.Format(CultureInfo.CurrentCulture, Resources.VsResources.Warning_ProjectDependencyMissingLowerBound,
```

Comment:
I'm keeping throw exception because this one code will throw uncontrolled exception(with not really helping message) just few lines down on line #88, so we better to throw controlled exception.

So far all excepted operations are succeeding with warning which is same as `msbuild` and `dotnet` works now.

Reply:
Does this change anything visually for the end user?

Reply:
I'm wondering if you accidentally discovered a repro for https://github.com/NuGet/Home/issues/10336. 
cc @dtivel @rrelyea @sbanni

Reply:
@nkolev92 

I removed this part. As you suspected it doesn't change anything for end user. Also it have very different error message from [NuGet/Home#10336](https://github.com/NuGet/Home/issues/10336).